pulsar/producer_partition.go

```diff
@@ -851,13 +851,16 @@ func (p *partitionProducer) internalSingleSend(mm *pb.MessageMetadata,
 	p._getConn().WriteData(buffer)
 }
 
+type pendingItemCallback func(err error)
+
 type pendingItem struct {
 	sync.Mutex
 	buffer       internal.Buffer
 	sequenceID   uint64
 	sentAt       time.Time
 	sendRequests []interface{}
 	completed    bool
+	callback     pendingItemCallback
```

Comment:
How about naming it as `flushCallback` since this callback is only used in the flush logic.

Reply:
OK, I will rename it next week.